datatypes/mcs_datatype.h

```diff
@@ -231,6 +231,16 @@ class SystemCatalog
           }
       }
 
+      /**
+          @brief The method returns true if the column precision
+          belongs to a wide decimal range.
+      */
+      inline bool isWideDecimalPrecision() const
```

Comment:
We can't trust precision detecting DECIMAL type b/c there is a division case( decimal(15) / decimal(38) ) that produces wide decimal with a precision < 18. I would suggest to use width as a marker in the context of this patch where it is possible.

Reply:
We are only using this function when the source type is a BIGINT/UBIGINT whose precision is sometimes set as 19. This causes issues when the caller detects that the value is wide decimal based on precision (as an example, in our implementations of the arithmetic operations in mcs_decimal.h). Width cannot be correctly used here since BIGINT/UBIGINT are 8 bytes.